src/modules/endpoint/sagas/watchFetchNews.js

```diff
@@ -39,27 +39,32 @@ export function * fetchNews (
       !isLocalNews
     )
     const languageValid = languages
-      .map(language => language.code)
-      .includes(language)
+      .find(languageModel => languageModel.code === language)
     const pushLanguagesAction = {
       type: 'PUSH_LANGUAGES',
       params: { languages: languages }
     }
     yield put(pushLanguagesAction)
 
     if (languageValid) {
-      const [newsList] = yield all([
+      const [news] = yield all([
```

Comment:
Wouldn't the following also work:

```
const news = isLocalNews ? yield call(loadLocalNews, city, language) : newsId ? yield call (loadTunewsElement, ...) : yield call(loadTuNews, ...)
```

Reply:
I.e. not sure why you need the `yield all` and why you only have a `yield`before `call(loadTunewsElement, ..)`

Reply:
yes would work.